### PR TITLE
Pedestal 0.4.0

### PR DESCRIPTION
**Pull request by aew**
I'm seeing the last test in `core-test` failing because schema validation doesn't seem to be happening for the `/doc` endpoint - seems related to the pedestal bump but not clear to me how - any ideas?

### CODE REVIEW

**Comment by frankiesardo** [comment]
Also, bumping to `0.4.0` knackered the sample project. I find it hard to understand what should be changed to make it work with the new version, a little bit more documentation wouldn't hurt.

Maybe @stuarth #2 can help?

**Comment by stuarth** [comment]
@frankiesardo that pr is a little out of date. i'll have a look at the sample project this afternoon though!

**Comment by aew** [comment]
The tests are now passing - I finally grokked https://github.com/pedestal/pedestal/pull/330 and updated the sample routes to something that is supported by the `:prefix-tree` router.  Specifying the `:linear-search` router also had tests passing without the route change.

File: sample/src/sample/service.clj [hunk 1]
```diff
@@ -219,10 +217,11 @@
         {:post add-order}
         ["/:id" ^:interceptors [load-order-from-db]
          {:get get-order-by-id}]]
+
        ["/secure" ^:interceptors [basic-auth] {:delete delete-db}]
 
        ["/doc" {:get [(swagger/swagger-doc)]}]
-       ["/*resource" {:get [(swagger/swagger-ui)]}]]]]))
+       ["/swagger-ui/*resource" {:get [(swagger/swagger-ui)]}]]]]))
```
Comment (frankiesardo) [inline sample/src/sample/service.clj]: So it's not possible to use splat params at the root of the project? That sucks, I liked the swagger ui mounted on "/"

File: test/pedestal/swagger/core_test.clj [hunk 2]
```diff
@@ -69,7 +69,7 @@
                          (coerce-params) (validate-response)
                          auth-middleware]
      {:get get-handler}
-     ["/:id" ^:interceptors [id-middleware]
+     ["/x/:id" ^:interceptors [id-middleware]
```
Comment (frankiesardo) [inline test/pedestal/swagger/core_test.clj]: Is it a conflict to the new router or just an arbitrary change?

File: test/pedestal/swagger/core_test.clj [hunk 3]
```diff
@@ -101,32 +101,33 @@
                             :responses {400 {} 500 {}}}}}]
     (is (= expected (doc/doc-routes routes)))))
 
-(def app (make-app {::bootstrap/routes (doc/inject-docs
+(def app (make-app {::bootstrap/router :prefix-tree ;; or :linear-search
+                    ::bootstrap/routes (doc/inject-docs
                                         {:title "Test"
```
Comment (frankiesardo) [inline test/pedestal/swagger/core_test.clj]: I forgot I should update this guy

**Comment by frankiesardo** [comment]
:+1: Wow thanks Edward. The PR looks great, and if you could squash it you would make it awesome ;)

File: sample/src/sample/service.clj (hunk 1 above)
Comment (stuarth) [inline sample/src/sample/service.clj]: yep, I mentioned it briefly on https://github.com/nubank/pedestal-swagger/pull/1

Comment (frankiesardo) [inline sample/src/sample/service.clj]: Is this a feature or a bug? I don't see any issues opened on pedestal about it.

Comment (aew) [inline sample/src/sample/service.clj]: it's a feature - you can use the old linear-search router if you want to retain the old behavior, but the prefix tree router does not support this (pedestal/pedestal#330)

File: test/pedestal/swagger/core_test.clj (hunk 2 above)
Comment (aew) [inline test/pedestal/swagger/core_test.clj]: conflict with the new router - using the linear search router this is not necessary

**Comment by aew** [comment]
Are there next steps / requests here?

**Comment by frankiesardo** [comment]
Cheers, I'll cut a new release soon

**Pull request merged**